File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    for i in s:
        if stack and i == ')' and stack[-1] == '(':
            stack.pop()
        else:
            stack.append(i)
    print(stack)
    if stack:
        return False
    else:
        return True

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        if(s.charAt(0) == ')')
            return false;
        for(int i=0; i<s.length(); i++){
            if(!(stack.isEmpty()) && s.charAt(i) == ')' && stack.peek().equals('(')){
                stack.pop();
            }
            else{
                stack.push(s.charAt(i));
            }
        }
        
        if(stack.isEmpty()){
            return true;
        }
        else
            return false;
    
    }
}